### PR TITLE
Support for the ICV nthreads-var in ompd_enumerate_icvs(), ompd_get_icv_from_scope() and ompd_get_icv_string_from_scope()

### PR DESCRIPTION
**Pull request by jinisusan**
Review and pull request for providing support for the ICV nthreads-var. (https://github.com/OpenMPToolsInterface/LLVM-openmp/issues/72).

Thanks!
Jini.

### CODE REVIEW

**Comment by jinisusan** [comment]
Gentle reminder ! 

Thanks,
Jini

**Comment by jprotze** [comment]
As I read the spec, `ompd_get_icv_from_scope` should always just provide a single integer in `icv_value` for which the tool provides the storage. If that is not possible, the function can return `ompd_rc_incompatible`.
In this case, the tool would need to use and parse `ompd_get_icv_string_from_scope`.

For _ompd-nthreads-var_ this means, the function can return a value if the variable is just a single value, but `ompd_rc_incompatible` for nested execution with a list of values.

Does this make sense?

**Comment by jinisusan** [comment]
Thank you very much, Joachim, for pointing this out. You are right -- this makes sense. Will send an updated revision.

**Comment by jinisusan** [comment]
Hi again Joachim, 

As I thought more about this, I feel that it might make more sense for the debugger to directly invoke ompd_get_icv_string_from_scope() for the ICVs like nthreads-var where the value of the ICV is a list (even if it constitutes of only a single element). This would avoid the (possibly) unnecessary calls from the debugger to ompd_get_icv_from_scope() for the cases where the value is a list of multiple elements. What do you think ? I am planning to ask about this in the omp-tools forum also. 

Thanks,
Jini.

**Comment by jinisusan** [comment]
Have committed changes according to Joachim's and John's suggestions in omp-tools. Please take a look.

Thanks!
Jini

**Comment by jinisusan** [comment]
Gentle reminder ! Thanks !

**Comment by jinisusan** [comment]
Made commits to replace an assert call with a ompd_rc_error return in the case of the callback table not being available (as  per the review comment in https://github.com/OpenMPToolsInterface/LLVM-openmp/pull/75 )

File: libompd/src/omp-icv.cpp [hunk 1]
```diff
@@ -236,6 +237,157 @@ static ompd_rc_t ompd_get_debug(
...
+    return ret;
+
+  /*__kmp_threads[t]->th.th_current_task->td_icvs.nproc*/
+  *nthreads_var_val = nproc;
+  /* If the nthreads-var is a list with more than one element, then the value of
+     this ICV cannot be represented by an integer type. In this case,
+     ompd_rc_incompatible is returned. The tool can check the return value and
+     can choose to invoke ompd_get_icv_string_from_scope() if needed. */
+  if (current_nesting_level < used - 1) {
+    return ompd_rc_incompatible;
```
Comment (jprotze) [inline libompd/src/omp-icv.cpp:319]: Note for OMPD specification: we might add a new return code: `ompd_rc_incomplete`.
For the currently available return codes, I would not expect meaningful values unless `ompd_rc_ok`.

**Comment by jinisusan** [comment]
Thanks, Joachim!

**Pull request merged**